### ISSUE TITLE
False positive "Namespace is never used"

### ISSUE THREAD

**Issue by emlyn** [jonase/eastwood#210]
When importing a class defined in another namespace with `defrecord`, it is necessary to first `require` the namespace so that the class is compiled before it can be imported (see https://clojuredocs.org/clojure.core/defrecord#example-542692d2c026201cdc326f87), but eastwood warns that the required namespace is never used.

**Comment by emlyn** [comment]
Actually a workaround for this is to set `:load-ns true` on the `defrecord` so that the `require` is not needed.

**Comment by dijonkitchen** [comment]
Any updates or should it be closed?

**Comment by jafingerhut** [comment]
There were some improvements made to Eastwood on its reporting of unused namespaces, but I do not recall whether those improvements fix this case, or not.

**Comment by mrkam2** [comment]
As of the "0.3.13" version, this is still an issue.

**Comment by mrkam2** [comment]
I added a draft PR with a testcase: https://github.com/jonase/eastwood/pull/351.

**Comment by mrkam2** [comment]
Are there any plans to address this issue?

**Comment by vemv** [comment]
(I'm merely an occasional contributor)

Looks like implementing a solution would entail creating a function akin to `protocols-used`, but for classes (namely those coming from defrecord + deftype) instead: https://github.com/jonase/eastwood/blob/7672782d748bcc15c0ea5bf1d913c3d47c9e0810/src/eastwood/linters/unused.clj#L269

Here is a sample `classes-used` implementation that worked for me just now, against your test case:

```clj
(defn classes-used [asts]
  (->> asts
       (mapcat ast/nodes)
       (keep :o-tag) ;; other "interesting" keys that can include a use class include: `:tag`, `:val`, `:result`
       set))
```

I don't have enough tools.analyzer expertise to tell which of the keys I have observed to be "interesting" (:o-tag, :tag, :val, :result) would be the most suitable for properly fixing this linter.

I'd encourage you to open a PR with a fix, spending a little more time in research than I did. Currently I lack the time/capability of opening proper PRs against Eastwood.

Cheers - V
